### PR TITLE
fix: mitigate blank screen during dev

### LINKED ISSUE #188: [Bug] Blank screen in dev
### Description

This bug happens when running UniMe in dev as desktop app when navigating in the credentials of the Ferris profile. Occasionally, I see a blank screen after some interactions.

### Hardware Specification

Operating system: macOS
RAM: 
Cores:
Device: 

### Steps to Reproduce the Bug

It's not reliably reproducible but it happens regularly. 

### Expected Behaviour

There should not be any blank screens.

### Actual Behaviour

To continue with dev, I can right-click and choose reload. It may take several reload attempts until the blank screen goes away. Occasionally, I need to restart the dev server.

### Errors

Vite CSS nesting error:

```sh
[vite:css] Nested CSS was detected, but CSS nesting has not been configured correctly.
Please enable a CSS nesting plugin *before* Tailwind in your configuration.
See how here: https://tailwindcss.com/docs/using-with-preprocessors#nesting
3  |      flex: 1;
4  |  
5  |      &:focus {
   |       ^
6  |        position: relative;
7  |      }
[vite:css] Nested CSS was detected, but CSS nesting has not been configured correctly.
Please enable a CSS nesting plugin *before* Tailwind in your configuration.
See how here: https://tailwindcss.com/docs/using-with-preprocessors#nesting
1  |  .trigger.s-W5HspqYnXoFG{flex:1;&:focus {
   |                                  ^
2  |        position: relative;
3  |      };&:focus-visible {
```

Vite dependency optimization error:

```sh
5:09:56 PM [vite] ✨ new dependencies optimized: @lottiefiles/lottie-player, tailwind-merge, @tauri-apps/api/path, @tauri-apps/plugin-fs
5:09:56 PM [vite] ✨ optimized dependencies changed. reloading
```

### PR DESCRIPTION
**Pull request by maiertech**
# Description of change

This PR addresses the blank screen issue that occurs frequently in dev when Vite optimizes dependencies and triggers a page reload.

## Links to any relevant issues

Fixes #188. Read the comments in this issue to understand the tradeoffs of this PR.

## How the change has been tested

You need to look at this PR as an attempt to make it unlikely that the blank screen issue occurs, and if it occurs to increase the chance that the app is in a state in which you can continue developing. High chance that the fixes do not cover all corner cases.

I tested by removing `node_modules` and installing dependencies several times, which should trigger Vite dependency optimizations. There should be none because of the dependency inclusion list in `vite.config.ts`.

## Definition of Done checklist

Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: unime/vite.config.ts [hunk 1]
```diff
@@ -44,5 +44,23 @@ export default defineConfig(async () => ({
   },
   optimizeDeps: {
     exclude: ['~icons/*'],
+    // #188: List of dependecies that Vite frequently optimizes.
+    // Use `include`, not `exclude`: https://github.com/sveltejs/kit/issues/11793#issuecomment-1965850225.
+    // TODO Check if this list can be removed after upgrading to Vite 5.
+    include: [
```
Comment (daniel-mader) [inline unime/vite.config.ts:50]: I found another two that optimized on my machine:
```
'date-fns',
'date-fns/locale'
```

Comment (daniel-mader) [reply unime/vite.config.ts:50]: Also this one:
```
@tauri-apps/plugin-barcode-scanner
```

Comment (maiertech) [reply unime/vite.config.ts:50]: Added these dependencies.

**Review by daniel-mader — APPROVED** [review]

**Review by nanderstabel — APPROVED** [review]

**Review by Oran-Dan — APPROVED** [review]
nice work, I've not had the blank screen once yet.

**Pull request merged**